src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -112,60 +115,88 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
                 return new ProjectPackages(Array.Empty<PackageReference>(), Array.Empty<TransitivePackageReference>());
             }
 
-            IList<LockFileTarget> targetsList = null;
+            IReadOnlyList<LockFileTarget> targetsList = null;
+            T installedPackages;
+            T transitivePackages;
             if (IsInstalledAndTransitiveComputationNeeded)
             {
                 // clear the transitive packages cache, since we don't know when a dependency has been removed
-                InstalledPackages.Clear();
-                TransitivePackages.Clear();
-                targetsList = await GetTargetsListAsync(assetsPath, token);
+                installedPackages = new T();
+                transitivePackages = new T();
+                targetsList = (await GetTargetsListAsync(assetsPath, token))?.ToList();
+            }
+            else
+            {
+                if (InstalledPackages == null || TransitivePackages == null)
```

Comment:
question: What's the scenario where these are both null and IsInstalledAndTransitiveComputationNeeded is false? 
Some sort of race condition?

Reply:
Most likely never happen, I added this just in case someone refactor code later and assign value one and don't assign the other.